src/types.rs

```diff
@@ -0,0 +1,37 @@
+//! Re-exports of types for glossing over no_std/std distinctions
```

Comment:
This whole module is a giant hack so we have a singular name we can use across both `no_std` and `std` to refer to these modules.